monitoring/bin/change_grafana_admin_password.sh

```diff
@@ -0,0 +1,133 @@
+  log_message  ""
+}
+
+# Assigning passed in parameters as variables for the script:
+POS_PARMS=""
+
+# Setting passed in variables:
+while (( "$#" )); do
+  case "$1" in
+    -ns|--namespace)
+      if [ -n "$2" ] && [ ${2:0:1} != "-" ]; then
+        tenantNS=$2
+        shift 2
+      else
+        log_error "A value for parameter [NAMESPACE] has not been provided." >&2
+        show_usage
+        exit 2
+      fi
+      ;;
+    -t|--tenant)
+      if [ -n "$2" ] && [ ${2:0:1} != "-" ]; then
+        tenant=$2
+        shift 2
+      else
+        log_error "A value for parameter [TENANT] has not been provided." >&2
+        show_usage
+        exit 2
+      fi
+      ;;
+    -p|--password)
+      if [ -n "$2" ] && [ ${2:0:1} != "-" ]; then
+        password=$2
+        shift 2
+      else
+        log_error "A value for parameter [PASSWORD] has not been provided." >&2
+        show_usage
+        exit 2
+      fi
+      ;;
+    -h|--help)
+      show_usage
+      exit
+      ;;
+    -*|--*=) # unsupported flags
+      log_error "Unsupported flag $1" >&2
+      show_usage
+      exit 1
+      ;;
+    *) # preserve positional arguments
+      POS_PARMS="$POS_PARMS $1"
+      shift
+      ;;
+  esac
+done
+
+# Convert namespace and tenant to all lower-case
+tenantNS=$(echo "$tenantNS"| tr '[:upper:]' '[:lower:]')
+tenant=$(echo "$tenant"| tr '[:upper:]' '[:lower:]')
+
+# Check for parameters - set with cluster or tenant
+if [ -z "$tenantNS" ] && [ -z "$tenant" ]; then
+  cluster="true"
+  namespace=$MON_NS
+  v4mSecret="v4m-grafana"
+elif [ -n "$tenantNS" ] && [ -n "$tenant" ]; then
+  cluster="false"
+  namespace=$tenantNS
+  v4mSecret="v4m-grafana-${tenant}"
+else
+  log_error "Both a [NAMESPACE] and a [TENANT] are required in order to change the Grafana admin password.";
+  exit 1
+fi
+
+# Check and make sure that a password was provided.
+if [ -z $password ]; then
+  log_error "A value for parameter [PASSWORD] has not been provided"
+  exit 1
+fi 
+
+if [ "$cluster" == "true" ]; then
+  grafanaPod="$(kubectl get pods -n $MON_NS -l app.kubernetes.io/name=grafana --template='{{range .items}}{{.metadata.name}}{{end}}')"
+else
+  grafanaPod="$(kubectl get pods -n $tenantNS -l app.kubernetes.io/name=grafana -l app.kubernetes.io/instance=v4m-grafana-${tenant} --template='{{range .items}}{{.metadata.name}}{{end}}')"
+fi
+
+# Error out if a Grafana pod has not been found
+if [ -z $grafanaPod ]; then
+  log_error "Unable to update Grafana password."
+  log_error "No Grafana pods were available to update"
+  exit 1
+fi
+
+# Changes the admin password using the Grafana CLI
+log_info "Updating the admin password using Grafana CLI"
+kubectl exec -n $namespace $grafanaPod -c grafana -- bin/grafana-cli admin reset-admin-password $password
+
+# Exit out of the script if the Grafana CLI call fails
+if (( $? == 1 )); then
+  log_error "An error occurred when updating the password"
```

Comment:
Does a failure always return a 1?  Or is any non-zero return code a sign of a problem?  Might want to use a check that the exit code is NOT 0.

Reply:
I'm not sure how else it could return so I'll switch it to NOT 0 like you suggested.